internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -313,6 +321,45 @@ func (c DockerCmdClient) IsContainerRunning(containerName string) (bool, error)
 	return output != "", nil
 }
 
+// KillContainer sends a SIGTERM signal to a Docker container with the given containerName if it's running.
+func (c DockerCmdClient) KillContainer(containerName string) error {
+	if running, err := c.containerStatus.IsContainerRunning(containerName); err != nil {
+		return err
+	} else if !running {
+		return nil
+	}
```

Comment:
I think it is possible to remove the else
```suggestion
running, err := c.containerStatus.IsContainerRunning(containerName)
if err != nil {
    return err
}

if !running {
    return nil
}

```